python/core-sdk/examples/data_product_generation/data_product_generation.py

```diff
@@ -0,0 +1,329 @@
+                   - ds: [123] for data source IDs
+                   - tag_ids: [456] for tag IDs
+                   - flags: ["Endorsement"] for endorsement flags
+                   - Any combination of the above
+        product_name: Name for the data product (e.g., "Sales Analytics")
+
+    Returns:
+        Generated YAML string for Alation Data Product
+    """
+
+    print(f"Creating '{product_name}' data product...")
+    print(f"Filtering by domain IDs: {domain_ids}")
+
+    # STEP 1: Fetch context (tables + SQL patterns in one go)
+    print("Step 1: Fetching catalog context...")
+
+    try:
+        # Get tables with relationships and columns
+        # Note: You can modify the search_filters below to use any supported criteria:
+        # - "ds": [123, 456] for data sources
+        # - "tag_ids": [789] for tags
+        # - "flags": ["Endorsement"] for endorsed content
+        # - Any combination of filters
+        # You could also add filters, sample values too if needed. We have kept it simple here.
+        tables = sdk.get_bulk_objects({
+            "table": {
+                "fields_required": ["name", "description", "common_joins", "columns"],
+                "search_filters": {
+                    "fields": {
+                        "domain_ids": domain_ids  # This could be any other search criteria
+                    }
+                },
+                "child_objects": {
+                    "columns": {"fields": ["name", "data_type", "description"]}
+                },
+                "limit": 15
+            }
+        })
+    except AlationAPIError as e:
+        print(f"Warning: Error fetching tables: {e}")
+        tables = {}
+
+    try:
+        # Get SQL patterns from the same scope
+        queries = sdk.get_bulk_objects({
+            "query": {
+                "fields_required": ["title", "description", "content"],
+                "search_filters": {
+                    "fields": {
+                        "domain_ids": domain_ids  # This could be any other search criteria
+                    }
+                },
+                "limit": 5
+            }
+        })
+    except AlationAPIError as e:
+        print(f"Warning: Error fetching queries: {e}")
+        queries = {}
+
+    # STEP 2: Get schema instructions from your Alation instance
+    print("Step 2: Getting latest schema instructions...")
+    try:
+        instructions = sdk.generate_data_product()
+        print(instructions)
+    except (AttributeError, AlationAPIError) as e:
+        print(f"Warning: Could not fetch schema from Alation instance: {e}")
+
+    # STEP 3: Generate YAML using AI model
+    print("Step 3: Generating data product YAML...")
+
+    # Create context for AI model
+    context_text = _format_context(tables, queries, product_name)
+
+    prompt = f"""
+{instructions}
+
+{context_text}
+
+Generate the YAML for "{product_name}" data product.
+"""
```

Comment:
This example is helpful! I initially thought the tool would accept the context and format it into the instruction, but this makes more sense. I hope this is clear to the customers too since the data product tool does not accept any parameters.